src/unity/toolkits/coreml_export/neural_net_models_exporter.cpp

```diff
@@ -291,10 +291,12 @@ std::shared_ptr<MLModelWrapper> export_activity_classifier_model(
   ModelDescription* model_desc = model.mutable_description();
 
   // Write the primary input features.
-  set_array_feature(model_desc->add_input(), "features",
-                    get_activity_classifier_features_short_description(
-                        prediction_window, features),
-                    { 1, prediction_window, features.size() });
+  for (size_t i = 0; i < features.size(); i++) {
+    set_array_feature(model_desc->add_input(), features[i],
+                      get_activity_classifier_features_short_description(
+                          prediction_window, features[i]),
+                      {1, prediction_window, features[i].size()});
```

Comment:
This doesn't look right: if we have e.g. six features, you're saying that _each_ feature should have input shape [1, 50, 6] (with prediction_window=50).

We should figure out whether the right shape is [1, 50, 1] or just a 1-d array [50]. (If we do the latter, we'd need to add some reshape into the exported model)

Reply:
Note the interface that appears in the screenshot here: https://apple.github.io/turicreate/docs/userguide/activity_classifier/export_coreml.html

It looks like stateIn and stateOut should appear as `(Double 400)`, but we probably want each named feature to similarly appear as `(Double 50)`. Then the developer doesn't have to worry about multi-arrays at all

Reply:
That sounds reasonable, this is why we wanted it to be named to eliminate multi-arrays. I think I can reshape it after and it should not be a problem. I will make appropriate changes.